src/main/java/app/model/agents/Grid.java

```diff
@@ -49,15 +49,31 @@ public Grid(double length, double width, CellType type, double cellSize)
 
     public Cell getCellAt(int row, int col)
     {
-        return grid[row][col];
+        try
+        {
+            Cell cell = grid[row][col];
+            return cell;
+        }
+        catch(Exception e)
+        {
+            System.out.println("Cell index out of bounds");
+        }
+        return null;
     }
 
     public Cell getCellAt(Vector position)
     {
-        int row = getCellRow(position);
-        int col = getCellCol(position);
-
-        return grid[row][col];
+        try
+        {
+            int row = getCellRow(position);
+            int col = getCellCol(position);
+            return grid[row][col];
+        }
+        catch(Exception e)
+        {
+            System.out.println("Cell index out of bounds");
+        }
+        return null;
```

Comment:
Same for exception here, I'm dubious if we could call this robust ...

Reply:
I wrote this to handle extreme edge cases, the only time this could occur if if the agent was looking for a cell outside the constraints of the map and therefore that direction is considered not viable, hence returning the null value, the print-out was to help me a bit, but is redundant